index.js

```diff
@@ -14,23 +14,20 @@ function SASSPlugin(optionsFn) {
 
 SASSPlugin.prototype.toTree = function(tree, inputPath, outputPath, inputOptions) {
   var options = Object.assign({}, this.optionsFn(), inputOptions);
-  var inputTrees;
-
-  if (options.onlyIncluded) {
-    inputTrees = [new Funnel(tree, {
-      include: ['app/styles/**/*'],
+  var ext = options.extension || 'scss';
+  var inputTrees = [
+    new Funnel(tree, {
+      // Only SASS files within the tree whose names match this expression will be copied,
+      // this avoid unnecessary SASS rebuild in the pipeline.
+      include: [options.onlyIncluded ? 'app/styles/**/*' : '**/*.' + ext],
```

Comment:
this will result in completely scanning all of `app/*` which can be quite slow

Reply:
If making it default is considerably slow, shall we at least leave it to an option to specify what to include/exclude would be great, coz along with our usages of ebryn/ember-component-css an couple of other plugins in the CSS pipeline, this will make rebuild significantly better.

Reply:
Isn't Funnel filtering supposed to be fast? I've seen filtering with file extension is quite a common tasks in several plugins.
https://github.com/ebryn/ember-component-css/blob/master/index.js#L22

Reply:
> Isn't Funnel filtering supposed to be fast? I've seen filtering with file extension is quite a common tasks in several plugins.

It's quick, but not faster then not filtering and only linking.

Reply:
My first thought is, shouldn't `SassCompiler` be doing this filtering?

Reply:
I have another project using ember-cli-sass, I'll try to find some time to investigate some as well.

Reply:
I thought the same, but...it seems like that depends on how [broccoli-sass-source-maps](https://github.com/blackbellme/broccoli-sass-source-maps/blob/master/index.js#L20) works.

Thanks for looking after this problem, it's a big pain for our rebuild here, I would provide some context:
```
ember-component-css@0.5.0
ember-cli-sass@7.0.0
ember-cli@2.16.2
ember-cli@
```

Reply:
@stefanpenner In case it isn't clear or hasn't been communicated, the project that @garryyao is working on is the Condé Nast Copilot Ember application (which I believe you've recently been given access to.

Reply:
@sohara ah, then that is the application I am also referring too :)

Reply:
Ha ha, cool. @garryyao works at the London office of Condé Nast International.

Reply:
fancy. Well I was actually going to be investigating exactly this for you guys anyways. I may come to the exact same conclusion. But I would feel more comfortable knowing this with confidence.

Reply:
Thanks Stefan, world is small!

We'd like to chase a bit after this since we feel a bit more confident about the fix we proposed here.

So in a project layout like ours, it make intensive use of in-project add-ons (lives under /lib) that's in pod structure (3 file types in a pod), Ember CLI will set those as [WatchedDir](https://github.com/ember-cli/ember-cli/blob/ee3f0685d84b7c63e02b8a0f6b6161a3535042a8/lib/models/addon.js#L437) and it's been included as [part of the css tree](https://github.com/ember-cli/ember-cli/blob/dbf9b0088afac665a2399d3b19df69b6d06e10dd/lib/broccoli/ember-app.js#L1456) which this plugin work against, will unnecessarily reload any plugin attached to part of the tree.